### DISCUSSION TITLE
how to redirect to default locale

### DISCUSSION THREAD

**Discussion by ghost2023** [amannn/next-intl#677] [Q&A]
I am migrating a large Nextjs 14 project to start using next-intl. how can i modify the middleware so that when going to '\users' it redirects it to the default locale with the same pathname like '\en\users'

**Comment by niksimFS24** [comment]
I am curious about this too!

**Comment by amannn** [comment]
That's a great question, thanks for bringing this up!

I've added a section in the docs about this in https://github.com/amannn/next-intl/commit/dbe192744cea996c465f567241993a585bfc7460, please see the updated [docs for `localePrefix: 'always'`](https://next-intl-docs.vercel.app/docs/routing/middleware#locale-prefix-always).

[The App Router example](https://next-intl-example-app-router.vercel.app/) uses this technique now too, you can try it with a request to https://next-intl-example-app-router.vercel.app/pathnames.

Hope this helps!

**Comment by niksimFS24** [comment]
I managed to get around my issues. Thanks for your time and efforts in making very good library @amannn :)
